src/format/CsvExporter.cpp

```diff
@@ -35,30 +35,44 @@ bool CsvExporter::exportDatabase(const QString& filename, const QSharedPointer<c
 
 bool CsvExporter::exportDatabase(QIODevice* device, const QSharedPointer<const Database>& db)
 {
-    QString header;
-    addColumn(header, "Group");
-    addColumn(header, "Title");
-    addColumn(header, "Username");
-    addColumn(header, "Password");
-    addColumn(header, "URL");
-    addColumn(header, "Notes");
-    header.append("\n");
+    if (device->write(exportHeader().toUtf8()) == -1) {
+        m_error = device->errorString();
+        return false;
+    }
 
-    if (device->write(header.toUtf8()) == -1) {
+    if (device->write(exportGroup(db->rootGroup()).toUtf8()) == -1) {
         m_error = device->errorString();
         return false;
     }
 
-    return writeGroup(device, db->rootGroup());
+    return true;
+}
+
+QString CsvExporter::exportDatabase(const QSharedPointer<const Database>& db)
```

Comment:
I think it would be better to just pass a QTextStream to the original exportDatabase function. This function does a completely different action with the same name.

Reply:
what about letting the client handle writing to a file if needed, and only return a `QString` when exporting the database from the `CsvExporter` and the `HtmlExporter`? Is there a reason I'm missing that would make it preferable to have the exporter work on streams instead of computing and returning the complete string value?

Reply:
Working on streams prevents the need to store additional information in RAM temporarily. It's not a deal breaker for our app, but in general streams offer more flexibility.